examples/src/bin/css.rs

```diff
@@ -0,0 +1,100 @@
+//! # CSS example
+//!
+//! This sample demonstrates how to use CSS with gtk-rs.
+
+use gio::prelude::*;
+use gtk::prelude::*;
+
+use std::env::args;
+
+use gdk::Display;
+use gtk::{
+    Application, ApplicationWindow, Box, Button, ComboBoxText, CssProvider, Entry, Orientation,
+    StyleContext, STYLE_PROVIDER_PRIORITY_APPLICATION,
+};
+
+// Basic CSS: we change background color, we set font color to black and we set it as bold.
+const STYLE: &str = "
+#entry1 {
+    background: linear-gradient(to right, #f00, #0f0);
+    color: blue;
+    font-weight: bold;
+}
+
+button {
+    /* If we don't put it, the yellow background won't be visible */
+    background-image: none;
+}
+#label1:hover {
+    transition: 500ms;
+    color: red;
+    background-color: yellow;
+}
+
+combobox button.combo box {
+    padding: 5px;
+}
+combobox box arrow {
+    -gtk-icon-source: -gtk-icontheme('pan-down-symbolic');
+    border-left: 5px solid transparent;
+    border-right: 5px solid transparent;
+    border-top: 5px solid black;
+}";
+
+fn build_ui(application: &Application) {
+    let window = ApplicationWindow::new(application);
+
+    window.set_title("CSS");
+
+    // The container container.
+    let vbox = Box::new(Orientation::Vertical, 0);
+
+    let label = Button::with_label("hover me!");
+    // We need to name it in order to be able to use its name as a CSS label to
+    // apply CSS on it.
+    WidgetExtManual::set_name(&label, "label1");
+
+    let entry = Entry::new();
+    // We need to name it in order to apply CSS on it.
+    WidgetExtManual::set_name(&entry, "entry1");
```

Comment:
Not necessarly, use entry.add_class(".entry1") instead and you can match it with css using `entry.entry1{ color: red;}`

Reply:
entry.add_class method does not exist. I see the same available in a StyleContext trait but button and entry both do not satisfy the same.
    let label = Button::with_label("hover me!");
    label.add_class("label1");

    let entry = Entry::new();
    entry.add_class(".entry1");
    entry.set_text("Some text");

Reply:
Oups, it's `add_css_class`